client/packages/requisitions/src/RnRForms/DetailView/RnRFormLine.tsx

```diff
@@ -310,57 +310,73 @@ export const RnRFormLine = ({
 const RnRNumberCell = ({
   value,
   disabled,
-  readOnly,
   onChange,
   textColor,
+  backgroundColor: inputBackgroundColor,
   max,
   error,
   allowNegative,
 }: {
   value: number;
   error?: boolean;
   disabled?: boolean;
-  readOnly?: boolean;
-  onChange: (val: number) => void;
+  onChange?: (val: number) => void;
   textColor?: string;
+  backgroundColor?: string;
   max?: number;
   allowNegative?: boolean;
 }) => {
-  const theme = useTheme();
-  const backgroundColor = readOnly ? theme.palette.background.drawer : 'white';
-
   const [buffer, setBuffer] = useBufferState<number | undefined>(
     NumUtils.round(value)
   );
 
+  const backgroundColor = inputBackgroundColor ?? 'white';
+
   return (
     <td style={{ backgroundColor }}>
       <Tooltip title={value === buffer ? '' : value}>
-        <NumericTextInput
-          InputProps={{
-            sx: {
-              backgroundColor,
-              '& .MuiInput-input': {
-                WebkitTextFillColor: textColor,
+        {disabled || !onChange ? (
+          <p
+            style={{
+              padding: '8px',
+              textAlign: 'right',
+              color: textColor,
+            }}
+          >
+            {buffer}
+          </p>
+        ) : (
+          <NumericTextInput
+            slotProps={{
+              input: {
+                sx: {
+                  backgroundColor,
+                  '& .MuiInput-input': {
+                    WebkitTextFillColor: textColor,
+                  },
+                },
               },
-            },
-          }}
-          error={error}
-          value={buffer}
-          disabled={readOnly ?? disabled}
-          onChange={newValue => {
-            setBuffer(newValue);
-            if (newValue !== undefined) onChange(newValue);
-          }}
-          max={max}
-          allowNegative={allowNegative}
-          defaultValue={0}
-          // NOTE: not setting input mode to text, because on Samsung tablets,
-          // the numeric keyboard doesn't allow entering negative numbers!
-          // Only needed for the negative columns, but better feel to have a consistent
-          // keyboard as you click through the whole R&R form
-          inputMode="text"
-        />
+              htmlInput: {
+                sx: {
+                  backgroundColor,
+                },
+              },
+            }}
+            error={error}
+            value={buffer}
+            disabled={disabled}
+            onChange={newValue => {
+              setBuffer(newValue);
+              if (newValue !== undefined) onChange(newValue);
+            }}
+            max={max}
+            allowNegative={allowNegative}
+            defaultValue={0}
+            // NOTE: setting input mode to text, because on Samsung tablets,
+            // the numeric keyboard doesn't allow entering negative numbers!
+            inputMode={allowNegative ? 'text' : 'numeric'}
```

Comment:
So now only for the negative field... if i remember right, this is how it was in the first place, and then bug was raised that the keyboard was wrong... but I think this is the best we can do 🤷